pandasdmx/reader/sdmxml.py

```diff
@@ -1208,15 +1208,9 @@ def parse_facet(self, elem):
         if isinstance(value_type, str):
             value_type = FacetValueType[value_type[0].lower() + value_type[1:]]

```

Comment:
This method is not self-explanatory and contains inefficient code. The copy and pop operations are not needed. Please explain why snake-case is to be returned.

Reply:
> The copy and pop operations are not needed.

The remaining elements of `attr` is used in the subsequent loop. The 'textType' XML attribute needs to be processed *first*, in order to create the Facet object, which is next populated by the loop. But `textType` is not guaranteed to be the first XML attribute appearing in `elem.attrib`; e.g. it could be the last. It is much simpler to pop this element from the dict and then loop over the remainder, than to handle it within a loop where it would appear in an unpredictable order.

`copy()` is needed because `elem.attrib.pop()` would destructively modify the XML document being read. This is not the expected behaviour of a "reader."

I will update/expand the comments. Since copy/pop are used consistently throughout the reader in this way, some of this can go in a file-level comment that explains the design.

> why snake-case is to be returned

All of the code follows [PEP 8 naming conventions](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables):

> **Method Names and Instance Variables**
>
> Use the function naming rules: lowercase with words separated by underscores as necessary to improve readability.

i.e., snake case. All of the attributes of the IM classes, including FacetType, follow PEP 8. SDMX-ML uses lower camel case for attribute names.